### PR TITLE
Delete IImplicitlyTriggeredBuildManager interface

### PR DESCRIPTION
**Pull request by mavasani**
Interface has been moved to CPS with [CPS#353192](https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/353192)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7640)

### CODE REVIEW

**Comment by mavasani** [comment]
@drewnoakes I will need your help in fixing the build errors that are coming up after moving to the latest CPS package.

**Review by shyamnamboodiripad — APPROVED** [review]

**Comment by drewnoakes** [comment]
> @drewnoakes I will need your help in fixing the build errors that are coming up after moving to the latest CPS package.

The issues here come from the addition of IVT in CPS. We now see all their internal types, and have conflicts.

Some of these are easy to address (nullable annotations). Others are not, where we have created our own copies of types that CPS has and modified them (i.e. DisposableBag). Apologies for not spotting this earlier.

I don't think we want IVT between CPS and the .NET PS. Can we just make the interface public on the CPS side?

**Comment by drewnoakes** [comment]
CPS PR https://dev.azure.com/devdiv/DevDiv/_git/CPS/pullrequest/353911 changes the interface to be public and obsolete, which will avoid the build issues here.

**Comment by mavasani** [comment]
Thanks @drewnoakes. Feel free to push a commit to this PR upgrading the package versions once https://dev.azure.com/devdiv/DevDiv/_git/CPS/pullrequest/353911 goes in.

**Comment by mavasani** [comment]
Upgraded the CPS package versions with latest commit - the new packages are not yet uploaded to the feed though, so the build will still fail. I'll requeue the build once the packages are available.

File: build/import/Packages.targets [hunk 1]
```diff
@@ -71,10 +71,10 @@
...
 
     <!-- CPS -->
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem"                                   Version="17.0.1036-pre" />
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Analyzers"                         Version="17.0.1036-pre" />
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Query"                             Version="17.0.1036-pre" />
-    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.SDK.Tools"                         Version="17.0.1036-pre" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem"                                   Version="17.0.1303-pre" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Analyzers"                         Version="17.0.1303-pre" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Query"                             Version="17.0.1303-pre" />
+    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.SDK.Tools"                         Version="17.0.1303-pre" />
```
Comment (mavasani) [inline build/import/Packages.targets:77]:
```suggestion
    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem"                                   Version="17.0.1304-pre" />
    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Analyzers"                         Version="17.0.1304-pre" />
    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.Query"                             Version="17.0.1304-pre" />
    <PackageReference Update="Microsoft.VisualStudio.ProjectSystem.SDK.Tools"                         Version="17.0.1304-pre" />
```

**Comment by mavasani** [comment]
@drewnoakes This should be ready for review.

**Review by drewnoakes — APPROVED** [review]

**Pull request merged**